### PR TITLE
fix(KongPlugin) compare CRD config and kongplugin config correctly

### LINKED ISSUE #102: The response-transformer plugin continuously detects outdated configuration to synchronize
### Summary

This has specifically been observed with the `response-transformer` plugin but may extend to others as well. 

In these cases, the `KongPlugin` resource is created (and attached to an Ingress), resulting in the entry correctly being added to Kong with a plugin associated to the route. The issue is that by watching the ingress-controller logs, you can see that the plugin is repeatedly trying to be updated.

The larger risk here is that it may trigger things like https://github.com/Kong/kong/pull/3423 this when many plugins are used.

**Kong Ingress controller version**

0.1.0

### Kubernetes version

1.9.3

### Environment

RDS Postgresql Kong Database

### What happened
The `KongPlugin` was created and the Kong configuration was created, but the ingress-controller logs show that it thinks the two are out of sync and tries to persist the changes every 10 minutes.

Immediately after saving a new configuration I can sometimes get a "up to date" message but on the next synchronization check (10 minutes?) it's back to being "outdated".

### Expected behvaior
Updates are not attempted against Kong when the KongPlugin and Kong database appear to be in sync.

### Steps To Reproduce

1. Create a KongPlugin with the following configuration
```
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: add-security-headers
config:
  add:
    headers:
    - "X-Xss-Protection:1; mode=block"
```

2. Attach this plugin to an Ingress using `response-transformer.plugin.konghq.com: add-security-headers`

3. Tail the `ingress-controller` container logs filtered for `add-security-headers`

4. Observe similar to the following logs that repeat regularly
```
I0823 01:27:55.081377       7 kong.go:784] configuring plugin stored in KongPlugin CRD add-security-headers
I0823 01:27:55.133977       7 kong.go:824] plugin add-security-headers configuration in kong is outdated. Updating...
```

### Other Information
I spent a bit of time trying to further isolate this without much success. I did try specifying all components of the response-transformer plugin (remove/append/replace) with empty lists/maps with no success. I'm having trouble determining what exactly is causing it to think the values are mismatched.

### PR DESCRIPTION
**Pull request by fffonion**
**What this PR does / why we need it**: Use `reflect.DeepEqual` to correctly compare CRD config and existing KongPlugin config

**Which issue this PR fixes** : fixes #102

### CODE REVIEW

File: internal/ingress/controller/kong_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	"testing"
+
+	kongadminv1 "github.com/kong/kubernetes-ingress-controller/internal/apis/admin/v1"
+)
+
+func TestPluginDeepEqual(t *testing.T) {
+	var equal bool
+	equal = pluginDeepEqual(map[string]interface{}{}, &kongadminv1.Plugin{Config: map[string]interface{}{}})
+	if !equal {
+		t.Errorf("Comparation empty maps  failed on pluginDeepEqual")
```
Comment (hbagdi) [inline internal/ingress/controller/kong_test.go]: s/Comparation/Comparing

There is an extra space between maps and failed.

Also, "on pluginDeepEqual" is not necessary here since Go's testing framework will communicate that `TestPluginDeepEqual` failed. Can we remove that and make the error line as:
> Comparing empty configs failed

Could you change it for other tests in this file? Thanks!

File: internal/ingress/controller/kong_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	}, &kongadminv1.Plugin{Config: map[string]interface{}{
+		"key3": "value3",
+		"key1": map[string]string{
+			"nestedkey1": "nestedvalue1",
+		},
+		"key4": "value4",
+	}})
+	if equal {
+		t.Errorf("Comparation maps with unmatched keys failed on pluginDeepEqual")
+	}
```
Comment (hbagdi) [inline internal/ingress/controller/kong_test.go]: Can we add two test cases:
1. Arrays comparison works as expected.
2. Ordering in array doesn't matter
3. What is the behavior for empty maps or array?

File: internal/ingress/controller/kong_test.go (hunk 1 above)
Comment (fffonion) [reply internal/ingress/controller/kong_test.go]: Added!

File: internal/ingress/controller/kong_test.go (hunk 2 above)
Comment (fffonion) [reply internal/ingress/controller/kong_test.go]: Added more tests to test nested array and the ordering, along with empty maps and arrays.
